imessage-exporter/src/exporters/txt.rs

```diff
@@ -276,9 +278,27 @@ impl<'a> Writer<'a> for TXT<'a> {
         _: &Message,
     ) -> Result<String, &'a str> {
         match &attachment.filename {
-            Some(filename) => Ok(filename.to_owned()),
+            Some(filename) => Ok(
+                if self.config.options.ios {
+                    // iOS rehash the filename to a path in the backup
+                    let salt = "MediaDomain-";
+                    let hash = format!("{:x}", Sha1::digest(
+                        format!("{}{}", salt, &filename[2..]).as_bytes()
```

Comment:
Use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get, don't directly index as it can panic.